### PR TITLE
[Fix] Education requirement in assessment dialog

### LINKED ISSUE #11531: 🐛 Education Requirement is not populated on Assessment Dialog
## 🐛 Bug

Education Requirement is not populated on Education Requirement Assessment Dialog for ALL candidates on the platform. 

## 🦋 Expected Behaviour

Education Requirement option and supporting evidence should be populated properly on the assessment dialog. 

## 🕵️ Details

Looks like front end issue as snapshot details are coming out properly from the backend. Snapshot download has it all. It is affecting all candidates on the platform and so should be looked into asap. 

## 📋 Steps to Reproduce

1. Open any process and select any candidate
2. Go to Screening and Assessment page and click on Education  Requirement to assess

## 📸 Screenshot

Image

## ✅ Acceptance Criteria

1. Education Requirement option and supporting evidence should be populated properly on the assessment dialog. 
2. Saving and editing Requirement Decision should be populated properly

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #11531 

## 👋 Introduction

Fixes the assessment status table dialog to show education requirement along with evidence.

## 🕵️ Details

> [!NOTE]
> This is not the best solution but for urgency, it works.

We were relying on the snapshot to render this but that information recently removed from the snapshot so no longer exists. This instead queries the info from the pool candidate directly and drills the required info down.

## 🧪 Testing

1. Build the app `pnpm run dev:fresh`
2. Login as admin `admin@test.com`
3. Apply to a process, taking note of the education requirement option/experience to select
4. Navigate to the admin view of the application
5. Open the education requirement assessment dialog
6. Confirm the option and experience you selected appears in the dialog

### CODE REVIEW

**Review by petertgiles — DISMISSED** [review]
Works great!  I actually don't think this is a bad solution.  I guess the prop drilling doesn't bother me as much.  :laughing:

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx
Comment (petertgiles) [inline apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:1]: Could you check on this story, please?
image

**Comment by esizer** [comment]
> Works great! I actually don't think this is a bad solution. I guess the prop drilling doesn't bother me as much. 😆

I guess it was more that I was not happy I had to do this kind of typing at a few levels :stuck_out_tongue: 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/16c92171bdfc3ec750d5b02f43e1c0dbedcdf881/apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx#L301-L305

https://github.com/GCTC-NTGC/gc-digital-talent/blob/16c92171bdfc3ec750d5b02f43e1c0dbedcdf881/apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx#L312-L319

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx
Comment (esizer) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:1]: Alright,  I fixed the story but in the process, I realized that the screening questions had the same issue so applied the same fix.

To test that, you will need to create a pool with an assessment step for screening questions to make sure the supporting evidence appears (question responses)

Comment (petertgiles) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:1]: Great!  Confirming that I can see the education requirement and question answers in the assessment dialogs.

**Review by petertgiles — APPROVED** [review]
:1st_place_medal:

**Pull request merged**